kafka/producer.go

```diff
@@ -17,4 +17,164 @@
 
 package kafka
 
-// TODO
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"go.uber.org/zap"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/plugin/kzap"
+
+	"github.com/elastic/apm-data/model"
+	"github.com/elastic/apm-queue/queuecontext"
+)
+
+type RecordMutator func(model.APMEvent, *kgo.Record) error
```

Comment:
Could you add a GoDoc comment here describing the purpose of an individual RecordMutator?

Reply:
Added!